src/screens/WalletConnect/CallRequest/SignatureRequestContent.js

```diff
@@ -47,10 +54,19 @@ type Props = {|
 function SignatureRequestContent({ request, onConfirm, onReject }: Props) {
   const { t } = useTranslation();
   const configs = useChainsConfig();
+  const activeAccount = useActiveAccount();
+  const isArchanovaAccountDeployed = useRootSelector(isArchanovaAccountDeployedSelector);
 
   const { title, iconUrl, chain } = getViewData(request);
   const config = configs[chain];
 
+  /**
+   * Archanova account needs to be deployed regardless of the action,
+   * Etherspot account must to be deployed for signature requests only
+   */
+  const requiresDeployedAccount = (isArchanovaAccount(activeAccount) && !isArchanovaAccountDeployed)
+    || isEtherspotAccountDeployed(activeAccount, chain);
+
```

Comment:
Important: second part of the condition seems to be incorrect

```suggestion
  const requiresDeployedAccount = (isArchanovaAccount(activeAccount) && !isArchanovaAccountDeployed)
    || (isEtherspotAccount(activeAccount) && !isEtherspotAccountDeployed(activeAccount, chain);
```

But probably you should use `isDeployedOnChainSelector` which contains the same logic in a nice encapsulation.

Reply:
No, because it returns hardcoded values for deployed states which won't work in case it's not actually deployed. Will add generic selector to make it more clear.